openedx/core/djangoapps/edx_global_analytics/tests/test_acceptor_api_usage.py

```diff
@@ -0,0 +1,173 @@
+        """
+        access_token_registration('https://mock-url.com')
+
+        mock_request.assert_called_once_with('https://mock-url.com' + '/api/token/registration/')
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.models.AccessTokensStorage.objects.create')
+    def test_saving_token(self, mock_access_tokens_storage_model_objects_create_method, mock_request):
+        """
+        Verify that access_token_registration gets access token after registration request to acceptor and
+        save it to AccessTokensStorage model in database.
+        """
+        mock_access_token = uuid.uuid4().hex
+
+        mock_request.return_value.json.return_value = {
+            'access_token': mock_access_token
+        }
+
+        access_token_registration('https://mock-url.com')
+
+        mock_access_tokens_storage_model_objects_create_method.assert_called_with(access_token=mock_access_token)
+
+    def test_token_authorization(self, mock_request):
+        """
+        Verify that access_token_authorization sends request to acceptor API for token authorization.
+        """
+        mock_access_token = uuid.uuid4().hex
+
+        access_token_authorization(mock_access_token, 'https://mock-url.com')
+
+        mock_request.assert_called_once_with(
+            'https://mock-url.com' + '/api/token/authorization/', data={'access_token': mock_access_token, }
+        )
+
+    @patch('openedx.core.djangoapps.edx_global_analytics.models.AccessTokensStorage.objects.first')
+    def test_refreshing_token(self, mock_access_tokens_storage_model_objects_first, mock_request):
+        """
+        Verify that access_token_authorization method gets refreshed access token after authorization request to
+        acceptor if Acceptor for any reason does not has a token from edX installation in own database.
+        """
+        mock_access_token = uuid.uuid4().hex
+        mock_refreshed_access_token = uuid.uuid4().hex
+
+        mock_request.status_code = 401
+
+        mock_request.return_value.json.return_value = {
+            'refreshed_access_token': mock_refreshed_access_token
+        }
+
+        access_token_authorization(mock_access_token, 'https://mock-url.com')
+
+        mock_access_tokens_storage_model_objects_first.access_token = mock_refreshed_access_token
+
+        self.assertEqual(mock_refreshed_access_token, mock_access_tokens_storage_model_objects_first.access_token)
+        mock_access_tokens_storage_model_objects_first.save.assert_called_once()
+
+    def test_sending_statistics(self, mock_request):
+        """
+        Verify that send_instance_statistics_to_acceptor sends request to acceptor API for token dispatch statistics.
+        """
+        send_instance_statistics_to_acceptor('https://mock-url.com', {'mock_data': 'mock_data', })
+
+        mock_request.assert_called_once_with(
+            'https://mock-url.com' + '/api/installation/statistics/', {'mock_data': 'mock_data', }
+        )
```

Comment:
> you call method explicitly with same data, that you testing in called_one. What is the reason of the test?)

I test, that `send_instance_statistics_to_acceptor` method get `X` argument and then make request with the same `X` parameter.